src/impl_/pymodule.rs

```diff
@@ -74,9 +82,22 @@ impl ModuleDef {
         let module = unsafe {
             Py::<PyModule>::from_owned_ptr_or_err(py, ffi::PyModule_Create(self.ffi_def.get()))?
         };
-        if self.initialized.swap(true, atomic::Ordering::SeqCst) {
+        let current_interpreter =
+            unsafe { ffi::PyInterpreterState_GetID(ffi::PyInterpreterState_Get()) };
```

Comment:
Looks like `PyInterpreterState_Get` was only added on Python 3.9. On 3.8 this was private `_PyInterpreterState_Get`.

So on 3.7 builds (or 3.8 abi3) we can't improve this check, I will adjust this PR to use the old one in those cases. The old check still works, it's just not as nice.